File: output/nemotron-vs-qwen35-ud/20260326-raw/executor-agentic/workspaces/agentic_python_metrics_contract/nemotron_iq4_xs/repeat-2/tests/test_metrics_contract.py
import unittest

from client.state import reduce_client_state
from runtime.metrics import MetricsTracker
from runtime.server import emit_state


class MetricsContractTests(unittest.TestCase):
    def test_metrics_live_in_session_state(self) -> None:
        tracker = MetricsTracker()
        tracker.turns_total = 9
        tracker.queued_actions = 3
        session_state, world_snapshot = emit_state("running", tracker.snapshot())
        self.assertEqual(world_snapshot["payload"].get("metrics"), None)
        self.assertEqual(session_state["payload"]["runtime"]["metrics"]["turns_total"], 9)
        self.assertEqual(session_state["payload"]["runtime"]["metrics"]["queued_actions"], 3)

    def test_client_reads_metrics_from_session_state(self) -> None:
        tracker = MetricsTracker()
        tracker.invalid_output_count = 4
        session_state, _ = emit_state("running", tracker.snapshot())
        state = reduce_client_state(session_state, {})
        self.assertEqual(state["runtime"]["metrics"]["invalid_output_count"], 4)


if __name__ == "__main__":
    unittest.main()

File: output/gemma4-vs-qwen35-ud/20260404-run/workspaces/agentic_ts_metrics_contract_patch/qwen35_a3b_ud_q4_xl/repeat-1/tests/test_metrics_contract.mjs
import test from "node:test";
import assert from "node:assert/strict";
import { buildMessages } from "../apps/game-runtime/src/runtime/sessionState.ts";
import { reduceClientState } from "../apps/game-client/src/state.ts";

test("metrics live in session.state runtime", () => {
  const metrics = {
    turnsTotal: 9,
    invalidOutputCount: 4,
    actionAppliedCount: 3,
    actionRejectedCount: 1,
    queuedActions: 2,
  };
  const { sessionState, worldSnapshot } = buildMessages(metrics);
  assert.equal(worldSnapshot.payload.metrics, undefined);
  assert.equal(sessionState.payload.runtime.metrics.turnsTotal, 9);
});

test("client reads metrics from session.state", () => {
  const metrics = {
    turnsTotal: 2,
    invalidOutputCount: 1,
    actionAppliedCount: 5,
    actionRejectedCount: 0,
    queuedActions: 3,
  };
  const { sessionState } = buildMessages(metrics);
  const next = reduceClientState({}, sessionState);
  assert.equal(next.runtime.metrics.queuedActions, 3);
});
